### DISCUSSION TITLE
Unable to start webrepl_setup

### DISCUSSION THREAD

**Discussion by IoTrees** [micropython/micropython#12483] [RP2040 / Pico] [answered]
Hello, 

I am just starting playing with the Wiznet w5100s with RP2040.

`help('modules')` lists webrepl and webrepl_setup

but when I connect to REPL via serial and try `import webrepl_setup` or put the import in my main.py, I get the following error:

`MPY: soft reboot
Traceback (most recent call last):     
  File "main.py", line 3, in <module>
  File "webrepl_setup.py", line 1, in <module>
  File "webrepl_setup.py", line 1, in main
  File "webrepl_setup.py", line 1, in get_daemon_status
OSError: [Errno 2] ENOENT
MicroPython v1.20.0 on 2023-04-26; W5100S-EVB-Pico with RP2040
Type "help()" for more information.  `

It seems like it does not see the files needed for webrepl_setup butI am thinking that I should not have to add any files [(see here)](https://github.com/orgs/micropython/discussions/10238) to use the webrepl_setup because its in my modules list.

Do I have things configured correctly?

**Comment by andrewleech** [comment, marked as answer]
It looks like webrepl_setup expects there to be a boot.py file already present on the filesystem in the current directory. Can you check if one exists on your board?
``` python
import os
os.listdir(".")
```

**Reply by IoTrees** [reply]
This was my problem, did not know I needed a boot.py for it to work

Thank you!
